lib/gui/components/drive-info-modal/templates/drive-info-modal.tpl.html

```diff
@@ -0,0 +1,59 @@
+<div class="modal-header">
+  <h4 class="modal-title">
+    <span class="glyphicon glyphicon-{{ ::modal.options.icon }}"></span>
+    <span>{{ ::modal.options.title }}</span>
+  </h4>
+  <button class="close" ng-click="modal.reject()">&times;</button>
+</div>
+
+<div class="modal-body">
+  <div class="modal-text"
+    ng-if="modal.options.header && modal.options.description">
+    <h4>{{ ::modal.options.header }}</h4>
+    <p>{{ ::modal.options.description }}</p>
+  </div>
+  <div class="modal-selected-drive">
+    <div class="modal-selected-drive-description">
+      {{ ::modal.selection.getDrive().description }}
+
+    </div>
+    <div class="modal-selected-drive-name">
+      {{ ::modal.selection.getDrive().name }} - {{ ::modal.selection.getDrive().size | gigabyte | number:1 }} GB
+
+
+      <span class="label label-warning"
+          ng-show="modal.selection.isDriveLargeEnough(modal.selection.getDrive()) && !modal.selection.isDriveLocked(modal.selection.getDrive()) && !modal.selection.isDriveSizeRecommended(modal.selection.getDrive())">
+          <i class="glyphicon glyphicon-warning-sign"></i>
+          NOT RECOMMENDED</span>
+
+      <span class="label label-danger"
+        ng-hide="modal.selection.isDriveLargeEnough(modal.selection.getDrive())">
+          <i class="glyphicon glyphicon-resize-small"></i>
+          TOO SMALL FOR IMAGE</span>
+
+      <span class="label label-danger"
+        ng-show="modal.selection.isDriveLocked(modal.selection.getDrive()) && modal.selection.isDriveLargeEnough(modal.selection.getDrive())">
```

Comment:
Also @jviotti it looks like the current logic is dictating that if a drive is too small, and it's locked, then the warning shown is "TOO SMALL FOR IMAGE".
IMHO the fact that it's locked, out to be displayed in preference to the fact that it's too small? Moving the categorisation logic outside of the template file will make it easier to make this logic more consistent, without this current mish-mash of some of the warning labels here using `ng-show` and some of them using `ng-hide`.